### PR TITLE
Strength and StrengthAV

### PR DESCRIPTION
**Pull request by egolol**

### CODE REVIEW

File: braph2genesis/measures/_Strength.gen.m [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+g = s.get('G');  % graph from measure class
+A = g.get('A');  % adjency matrix (for graph) or 2D-cell array (for multiplex)
+
+strength = cell(g.layernumber(), 1);
+for li = 1:1:g.layernumber()
+    if g.is_graph(g)
+        Aii = A;
+    else
+        Aii = A{li, li};
+    end
```
Comment (annacanal) [inline braph2genesis/measures/_Strength.gen.m:35]: we don't need this if now.

just:
 Aii = A{li, li};

File: braph2genesis/measures/_StrengthAV.gen.m [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+%%% ¡compatible_graphs!
+GraphWU
+MultiplexGraphWU
+
+%% ¡props_update!
+
+%%% ¡prop!
+M (result, cell) is the average strength.
+%%%% ¡calculate!
+strength_av = calculateValue@Strength(m, prop);
```
Comment (annacanal) [inline braph2genesis/measures/_StrengthAV.gen.m:26]: call this strength

File: braph2genesis/measures/_StrengthAV.gen.m [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+
+g = m.get('G');  % graph from measure class
+
+strength_av = cell(g.layernumber(), 1);
+for li = 1:1:length(strength_av)
+    strength_av(li) = {mean(strength_av{li})};
```
Comment (annacanal) [inline braph2genesis/measures/_StrengthAV.gen.m:32]: strength_av(li) = {mean(strength{li})};

**Review by annacanal — CHANGES_REQUESTED** [review]

File: braph2genesis/measures/_Strength.gen.m [hunk 4]
```diff
@@ -30,11 +30,7 @@
 
 for li = 1:1:g.layernumber()
     A = cell2mat(A);
```
Comment (annacanal) [inline braph2genesis/measures/_Strength.gen.m:32]: this line shouldn't be there. (A = cell2mat(A);)
we want to treat them now always with cell, like in multiplex.
Aii = A{li, li}; in this way, it goes to the diagonal of supra-adjacency matrix : A{1, 1}, A{2,2}, ..A{L, L}

Check _Degree, should be the same inside the loop, without binarizing

**Review by c-yuwei — APPROVED** [review]

**Review by annacanal — APPROVED** [review]

File: braph2genesis/measures/_StrengthAv.gen.m [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+GraphWU
+%%%% ¡code!
+B = [
+    0  .2 1
+    .2 0  0
+    1  0  0
+    ];
+known_strength_av = {mean([1.2 0.2 1])'};
+g = GraphWU('B', {B});
+s_outside_g = Strength('G', g);
```
Comment (c-yuwei) [inline braph2genesis/measures/_StrengthAv.gen.m:49]: should be s_outside_g = StrengthAv('G', g);

File: braph2genesis/measures/_StrengthAv.gen.m [hunk 6]
```diff
@@ -0,0 +1,86 @@
...
+    B21     B22
+    };
+
+known_strength_av = {
+                 mean([1.2 .2  1]')
+                 mean([1   1.4 .4]')
+                 };
+                                
+g = MultiplexGraphWU('B', {B});
+strength_av = Strength('G', g);
```
Comment (c-yuwei) [inline braph2genesis/measures/_StrengthAv.gen.m:82]: Should be strength_av = StrengthAv('G', g);

**Review by c-yuwei — CHANGES_REQUESTED** [review]

File: braph2genesis/measures/_StrengthAv.gen.m (hunk 6 above)
Comment (annacanal) [reply braph2genesis/measures/_StrengthAv.gen.m:82]: Hi now I see this error on line 81.
When multiplex: g = MultiplexGraphWU('B', B);
if not the test won't pass

File: braph2genesis/measures/_Strength.gen.m [hunk 7]
```diff
@@ -0,0 +1,87 @@
...
+    B11     B12  
+    B21     B22
+    };
+
+known_strength = {
+                 [1.2 .2  1]'
+                 [1   1.4 .4]'
+                 };
+                                
+g = MultiplexGraphWU('B', {B});
```
Comment (annacanal) [inline braph2genesis/measures/_Strength.gen.m:82]: g = MultiplexGraphWU('B', B);
Here B is already a cell

File: braph2genesis/measures/_StrengthAv.gen.m [hunk 8]
```diff
@@ -0,0 +1,86 @@
...
+%%%% ¡name!
+GraphWU
+%%%% ¡code!
+B = [
+    0  .2 1
+    .2 0  0
+    1  0  0
+    ];
+known_strength_av = {mean([1.2 0.2 1])'};
+g = GraphWU('B', {B});
```
Comment (annacanal) [inline braph2genesis/measures/_StrengthAv.gen.m:48]: g = MultiplexGraphWU('B', B);

**Review by annacanal — CHANGES_REQUESTED** [review]

File: braph2genesis/measures/_StrengthAv.gen.m (hunk 6 above)
Comment (egolol) [reply braph2genesis/measures/_StrengthAv.gen.m:82]: ugg forgot to change this.... sometimes i change stuff in the tmp, and forget to change it at the genesis

**Review by annacanal — APPROVED** [review]

**Review by c-yuwei — APPROVED** [review]

**Review by giovannivolpe — APPROVED** [review]

**Pull request merged**